test/package/test_dependency_api.py

```diff
@@ -224,13 +224,9 @@ def test_intern_error(self):
 
         buffer = BytesIO()
 
-        try:
+        with self.assertRaisesRegex(PackagingError, "did not match against any action"):
```

Comment:
This assertion is not exactly the same as the one that was removed. It would be good to also check that the packages listed as unhandled are the ones we expect.